src/AutoRest.CSharp/Common/Output/Models/Types/TypeProvider.cs

```diff
@@ -45,7 +46,7 @@ protected TypeProvider(string defaultNamespace, SourceInputModel? sourceInputMod
         private TypeDeclarationOptions BuildType()
         {
             return BuilderHelpers.CreateTypeAttributes(
-                DefaultName,
+                DefaultName.ToCleanName(),
```

Comment:
This is wrong, please call ToCleanName in ModelTypeProvider:
https://github.com/Azure/autorest.csharp/blob/8671af7f3cb128ec8e9f18926833c3e15949d473/src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProvider.cs#L74